databases.go

```diff
@@ -233,3 +234,65 @@ func (c *Client) DatabaseDisableFeature(ctx context.Context, app, addonID, featu
 
 	return res, nil
 }
+
+type DbmsAttributes struct {
+	PasswordEncryption string `json:"password_encryption"`
+}
+
+type DatabaseUser struct {
+	Name           string          `json:"name"`
+	ReadOnly       bool            `json:"read_only"`
+	Protected      bool            `json:"protected"`
+	Password       string          `json:"password,omitempty"`
+	DbmsAttributes *DbmsAttributes `json:"dbms_attributes,omitempty"`
+}
+
+type DatabaseCreateUserParam struct {
+	DatabaseID           string `json:"database_id"`
+	Name                 string `json:"name"`
+	ReadOnly             bool   `json:"read_only"`
+	Password             string `json:"password,omitempty"`
+	PasswordConfirmation string `json:"password_confirmation,omitempty"`
+}
+
+type databaseCreateUserPayload struct {
+	DatabaseUser DatabaseCreateUserParam `json:"database_user"`
+}
+
+// DatabaseUserResponse is the response body of database create user
+type DatabaseUserResponse struct {
+	DatabaseUser DatabaseUser `json:"database_user"`
+}
+
+// DatabaseCreateUser creates an user to the given database addon
+func (c *Client) DatabaseCreateUser(ctx context.Context, app, addonID string, user DatabaseCreateUserParam) (DatabaseUser, error) {
+	res := DatabaseUserResponse{}
+	payload := databaseCreateUserPayload{
+		DatabaseUser: user,
+	}
+	err := c.DBAPI(app, addonID).SubresourceAdd(ctx, "databases", addonID, "users", payload, &res)
+	if err != nil {
+		return res.DatabaseUser, errors.Wrapf(ctx, err, "create a user on database %s", addonID)
+	}
+	return res.DatabaseUser, nil
+}
+
+// DatabaseUsersResponse is the response body of database list users
+type DatabaseUsersResponse struct {
+	DatabaseUsers []DatabaseUser `json:"database-users"`
+}
+
+// DatabaseListUsers list the users of the given database addon
+func (c *Client) DatabaseListUsers(ctx context.Context, app, addonID string) ([]DatabaseUser, error) {
+	res := DatabaseUsersResponse{}
+	err := c.DBAPI(app, addonID).SubresourceList(ctx, "databases", addonID, "users", nil, &res)
+	if err != nil {
+		return res.DatabaseUsers, errors.Wrap(ctx, err, "get database list of users")
```

Comment:
Hi there,
I noticed that you now use `errors.Wrap` instead of `errgo.Notef` elsewhere in the code. Is it expected ? This require a little bit of work when using theses endpoints when there is an error (and create some headache).

Reply:
Hi François,
We depreciate the usage of the package [errgo](https://pkg.go.dev/gopkg.in/errgo.v2) in general in favor of [errors](https://pkg.go.dev/github.com/pkg/errors). This latter is not from standard library, because it doesn't implement wrapping system yet.
But you have probably noticed that it's not the latter package itself that is used here and more generally in our code base.

We are wrapping these two first packages to be retro compatible during the transition and to take context (`ErrCtx`) by using our own package: ["github.com/Scalingo/go-utils/errors/v2"](https://pkg.go.dev/github.com/Scalingo/go-utils/errors/v2@v2.3.0).

If by _little bit of work_ you mean compare errors. I'd strongly advise you to use `Is` and/or `As` methods from ErrCtx, you can find them here:
https://github.com/Scalingo/go-utils/blob/77ec9e608897147630a4e56912c9318aa3d4c692/errors/cause.go#L10-L38
Their behavior is based on standard library.


FYI taking the context allows us to retrieve the root logger containing all fields, e.g. in middlewares.